src/main/java/seedu/jarvis/storage/JsonAdaptedMasteryCheck.java

```diff
@@ -20,4 +30,47 @@ public JsonAdaptedMasteryCheck(@JsonProperty("studentName") String studentName,
         this.dateAndTime = dateAndTime;
         this.passed = passed;
     }
+
+
+    /**
+     * Converts a given {@code MasteryCheck} into this class for Jackson use.
+     */
+    public JsonAdaptedMasteryCheck(MasteryCheck source) {
+        studentName = source.getStudentName();
+        dateAndTime = source.getDateAndTime().toString();
+        passed = source.isPassed() ? "T" : "F";
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted Mastery Check object into the model's {@code MasteryCheck} object.
+     * @return MasteryCheck
+     * @throws IllegalValueException if there were any data constraints violated in the adapted Mastery Check.
+     */
+    public MasteryCheck toModelType() throws IllegalValueException {
```

Comment:
Thank you for implementing this method, we will be able to save and retrieve MCs now!